src/compas/geometry/primitives/polyline.py

```
@@ -412,4 +412,74 @@ def divide_polyline_by_length(self, length, strict=True, tol=1e-06):
 
         return division_pts
 
+    def extend(self, length, both_ends=False):
+        """Extends a polyline from its start or end by a given length
+
+        Parameters:
+        -----------
+        length: list of float [start_offset, end_offset]
+
+        both_ends: bool
+            If True, both ends of Polyline will be extended
+
+        Returns
+        -------
+        Polyline
+        """
+        if not isinstance(length, list):
+            length = [length, length]
```

Not sure if this is such a great idea, I would stick to more strict input: it's either a list or we let it fail.